source/common/quic/envoy_quic_server_session.h

```diff
@@ -17,6 +17,14 @@
 namespace Envoy {
 namespace Quic {
 
+#define QUIC_CONNECTION_STATS(COUNTER)                                                             \
```

Comment:
This seems fine to me but I wonder if we should loop in @jmarantz for a double-check?

Reply:
Mostly: can we throw these names into QuicStatNames so we don't need to take a global symbol table lock whenever an EnvoyQuicServerSession is created.

I'm not sure how often that happens, but it sounds like it happens on the data-plane so it'd be better not to take what's effectively a singleton  lock at that point.

Reply:
I'm confused about how to add stats. Is macros like `MY_COOL_STATS(COUNTER, GAUGE, HISTOGRAM)` still the common way of doing that? QuicStatNames currently has stats for QUIC errors which are big enums and we want to lazy-initialize them. But the stats here only has a few counters.

If the concern is initializing this stats once per connection, is moving it up to be a listener stats more reasonable? Since other than the prefix is "quic.connection", there is no per-connection information in the stats name.

Reply:
I moved the creation of QuicConnectionStats into EnvoyQuicDispatcher constructor. PTAL!

Reply:
Thanks; looks good! I'm assuming here that (a) EnvoyQuicServerSession is only created on startup or config update and (b) the new stats are useful enough you want to instantiate them up front, rather than on demand.

If so, feel free to resolve; lgtm. I didn't look at the rest of the PR.

Reply:
> I'm assuming here that (a) EnvoyQuicServerSession is only created on startup or config update 

EnvoyQuicServerSession is not. But I think you meant EnvoyQuicDispatcher which instantiates those stats now. And yes, it is only created on startup or LDS update.